pkg/extension/wasmimage/wasmimage.go

```diff
@@ -0,0 +1,62 @@
+package wasmimage
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	orascnt "github.com/deislabs/oras/pkg/content"
+	ocispec "github.com/opencontainers/image-spec/specs-go/v1"
+)
+
+type WasmImage struct {
```

Comment:
You can ignore this because it is just personal comment and may have wrong consideration.
I think `WasmImage` should be stateful at the global sight. For more detail, it should have `pushed`, `pending`, `not pushed` state. So I consider that we should have `WasmImage` interface and implement `push()` method, with injecting `Pusher` dependency, and also we can have `Pusher` interface, right? If I have wrong pointing out about this, please notify it. Thanks.

Reply:
> I think WasmImage should be stateful at the global sight. For more detail, it should have pushed, pending, not pushed state.

oh, yes, that seems better in terms of their status managements. But currently, I don't expect to push multiple images at the same time, so we might not need to do it because the job works synchronously. what do you think about that?

Reply:
The indication signal may be used handling signals like `SIGTERM`, which will execute fallback. I'm not sure about the detail of the library which you used, but if the library doesn't have the handling mechanism, we may implement this feature. But it can't be used in this PR.